src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -87,6 +74,23 @@ void Terminal::Create(COORD viewportSize, SHORT scrollbackLines, Renderer& rende
     const TextAttribute attr{};
     const UINT cursorSize = 12;
     _mainBuffer = std::make_unique<TextBuffer>(bufferSize, attr, cursorSize, true, renderer);
+
+    auto dispatch = std::make_unique<AdaptDispatch>(*this, renderer, _renderSettings, *_terminalInput);
+    auto engine = std::make_unique<OutputStateMachineEngine>(std::move(dispatch));
+
+    auto& dispatchRef = engine->Dispatch();
+    _pfnSetCursorStyle = [&](auto style) { dispatchRef.SetCursorStyle(style); };
+    _pfnEraseScrollback = [&]() { dispatchRef.EraseInDisplay(DispatchTypes::EraseType::Scrollback); };

```

Comment:
If you add a `AdaptDispatch* _dispatchRef` member you can write
```cpp
_dispatchRef = dispatch.get();
// ...
if (_dispatchRef)
{
    _dispatchRef.SetCursorStyle(style);
}
// ...
if (_dispatchRef)
{
    _dispatchRef.EraseInDisplay(DispatchTypes::EraseType::Scrollback);
}
```

Design-wise, since we're now holding a reference to the dispatch class ourselves, we might actually hold the owning pointer and only give `OutputStateMachineEngine` a reference...

Reply:
This is actually the way I had it initially, but it felt wrong to me hanging on to the pointer like that. I wasn't particularly happy with the current solution either, so I don't mind changing it back if that's what you prefer.

The other option I considered was using a `reinterpret_cast` to get the `OutputStateMachineEngine` from the `_stateMachine`, and from there you can easily get the dispatcher. Something like this:

    auto& engine = reinterpret_cast<OutputStateMachineEngine&>(_stateMachine->Engine());
    engine.Dispatch().SetCursorStyle(cursorStyle);

I like that it doesn't require storing any additional members, but the `reinterpret_cast` seems dodgy to me. I'm not sure that's valid, but we are using that exact pattern elsewhere in the code.

As for holding on to the pointer and just giving the `OutputStateMachineEngine` a reference, I considered that too, but we're instantiating `OutputStateMachineEngine` in a whole lot of places, and having to update all of them seemed like a lot of work. Maybe I'm just being lazy though.

But as I say, I'm happy to go with whichever option you prefer, because I can't make up my mind one way or the other. I just wanted to bring up that `reinterpret_cast` idea first in case you like it.

Reply:
I just remembered there was another reason for not wanting to mess with the `OutputStateMachine` construction too much, and that is because conhost does some weird things with the state machine when you switch to the alt buffer, and I don't want to risk breaking that.